sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -139,6 +141,11 @@ object Literal {
    * Constructs a Literal from a String
    */
   def fromString(str: String, dataType: DataType): Literal = {
+    def parse[T](f: UTF8String => Option[T]): T = {
+      f(UTF8String.fromString(str)).getOrElse {
+        throw new AnalysisException(s"Cannot parse the ${dataType.catalogString} value: $str")
```

Comment:
Looks this can be called somewhere else not by the parser .. in that case, parse exception might not make much sense. It maybe has to be illigal argument exception or runtime exception as well.